server/service/src/invoice/prescription/update/validate.rs

```diff
@@ -0,0 +1,104 @@
+use crate::invoice::{
+    check_invoice_exists, check_invoice_is_editable, check_invoice_type, check_status_change,
+    check_store,
+};
+use crate::validate::{check_other_party, CheckOtherPartyType, OtherPartyErrors};
+use repository::{ClinicianRow, ClinicianRowRepository, EqualFilter, RepositoryError};
+use repository::{
+    InvoiceLineFilter, InvoiceLineRepository, InvoiceLineRowType, InvoiceRow, InvoiceRowStatus,
+    InvoiceRowType, StorageConnection,
+};
+
+use super::{UpdatePrescription, UpdatePrescriptionError};
+
+pub fn validate(
+    connection: &StorageConnection,
+    store_id: &str,
+    patch: &UpdatePrescription,
+) -> Result<(InvoiceRow, bool), UpdatePrescriptionError> {
+    use UpdatePrescriptionError::*;
+
+    let invoice = check_invoice_exists(&patch.id, connection)?.ok_or(InvoiceDoesNotExist)?;
+    if !check_store(&invoice, store_id) {
+        return Err(NotThisStoreInvoice);
+    }
+    if !check_invoice_is_editable(&invoice) {
+        return Err(InvoiceIsNotEditable);
+    }
+    if !check_invoice_type(&invoice, InvoiceRowType::Prescription) {
+        return Err(NotAPrescription);
+    }
+    if let Some(clinician_id) = &patch.clinician_id {
+        check_clinician_exists(connection, clinician_id)?.ok_or(ClinicianDoesNotExist)?;
+    }
+
+    // Status check
+    let status_changed = check_status_change(&invoice, patch.full_status());
+    if status_changed {
+        check_can_change_status_to_picked(connection, &invoice, patch.full_status())?;
+    }
+
+    if let Some(patient_id) = &patch.patient_id {
+        check_other_party(
+            connection,
+            store_id,
+            &patient_id,
+            CheckOtherPartyType::Patient,
+        )
+        .map_err(|e| match e {
+            OtherPartyErrors::OtherPartyDoesNotExist => OtherPartyDoesNotExist {},
+            OtherPartyErrors::OtherPartyNotVisible => OtherPartyNotVisible,
+            OtherPartyErrors::TypeMismatched => OtherPartyNotAPatient,
+            OtherPartyErrors::DatabaseError(repository_error) => DatabaseError(repository_error),
+        })?;
+    }
+
+    Ok((invoice, status_changed))
+}
+
```

Comment:
Can probably simplify this by passsing optional clinician id:
https://github.com/openmsupply/open-msupply/compare/1951-Endpoint-Update-Prescription...1951-Endpoint-Update-Prescription-Simplify-clinican-validation